tests/unit/cluster/atomic-slot-migration.tcl

```diff
@@ -149,61 +150,64 @@ start_cluster 3 3 {tags {external:skip cluster}} {
+            # for `streaming-buffer` and `send-bulk-and-stream` state, we need to set
+            # a longer delay to write incremental data to the main channel
+            if {$state eq "streaming-buffer" || $state eq "send-bulk-and-stream"} {
+                R 1 config set rdb-key-save-delay 1000000
+            }
 
             # Start the slot 0 write load on the R 1
             if {$::tls} { set port [lindex [R 1 config get tls-port] 1]
             } else { set port [lindex [R 1 config get port] 1] }
             set load_handle [start_write_load "127.0.0.1" $port 100 "06S"]
 
-            # Set the fail point for the main channel
+            # clear old fail points and set the new fail point
+            assert_equal {OK} [R 0 debug asm-failpoint "" ""]
+            assert_equal {OK} [R 1 debug asm-failpoint "" ""]
             if {$operation eq "import"} {
                 assert_equal {OK} [R 0 debug asm-failpoint "import-$channel-channel" $state]
-                assert_equal {OK} [R 1 debug asm-failpoint "" ""] ;# Clear migrate node fail point
             } elseif {$operation eq "migrate"} {
-                assert_equal {OK} [R 0 debug asm-failpoint "" ""] ;# Clear import node fail point
                 assert_equal {OK} [R 1 debug asm-failpoint "migrate-$channel-channel" $state]
             } else {
                 fail "Unknown operation: $operation"
             }
 
-            # Migrate slot 0-100 to R 0 if this is the first time,
-            # otherwise, import will retry automatically
-            if {$send_migration_import_0_100 == 0} {
-                R 0 CLUSTER MIGRATION IMPORT 0 100
-                set send_migration_import_0_100 1
-            }
+            # Start the migration
+            set task_id [R 0 CLUSTER MIGRATION IMPORT 0 100]
 
             # The task should be failed due to the fail point
             wait_for_condition 1000 50 {
                 [string match -nocase "*$channel*${state}*" [migration_status 0 0-100 error]] ||
                 [string match -nocase "*$channel*${state}*" [migration_status 1 0-100 error]]
             } else {
                 fail "ASM task did not fail with expected error -
-                     (dst: [migration_status 0 0-100 error],
-                      src: [migration_status 1 0-100 error],
+                     (dst: [migration_status 0 0-100 error]
+                      src: [migration_status 1 0-100 error]
                       expected: $channel $state)"
             }
             R 1 config set rdb-key-save-delay 0
             stop_write_load $load_handle
+
+            # Cancel the task
+            assert_equal {1} [R 0 CLUSTER MIGRATION CANCEL ID $task_id]
+            catch {R 1 CLUSTER MIGRATION CANCEL ID $task_id} err
+            # err is 1 or 'No ASM task found' since the task may not start on the source
+            if { $err ne "1" && ![string match -nocase "*No ASM task found*" $err]} {
```

Comment:
for the return value of MIGRATION CANCEL, maybe return 0 if task is not found?

Reply:
yeah, it makes sense.

Reply:
OK, let me change it